### PR TITLE
Refactored ext:dev:upload and switched to Publisher API.

### PR DESCRIPTION
**Pull request by apascal07**

### CODE REVIEW

File: src/commands/ext-dev-publish.ts [hunk 1]
```diff
@@ -16,14 +16,11 @@ marked.setOptions({
...
-  .option(
-    `--repo <repo>`,
-    `Public Git repo URI (only required for first version from repo, cannot be changed)`
-  )
+  .option(`--repo <repo>`, `Public GitHub repo URI that contains the extension source`)
   .option(`--ref <ref>`, `commit hash, branch, or tag to build from the repo (defaults to HEAD)`)
   .option(
     `--root <root>`,
-    `root directory that contains this Extension (defaults to previous version's root or root of repo if none set)`
+    `root directory that contains this extension (defaults to last uploaded root or repo root if none set)`
```
Comment (joehan) [inline src/commands/ext-dev-publish.ts:23]:
```suggestion
    `root directory that contains this extension (defaults to last uploaded root or "/" if none set)`
```

File: src/extensions/extensionsApi.ts [hunk 2]
```diff
@@ -19,7 +19,11 @@ import {
 const VERSION = "v1beta";
 const PAGE_SIZE_MAX = 100;
 
-const apiClient = new Client({ urlPrefix: extensionsOrigin, apiVersion: VERSION });
+const extensionsApiClient = new Client({ urlPrefix: extensionsOrigin, apiVersion: VERSION });
+const extensionsPublisherApiClient = new Client({
+  urlPrefix: extensionsOrigin,
+  apiVersion: VERSION,
```
Comment (joehan) [inline src/extensions/extensionsApi.ts:25]: Let's have separate consts for EXTENSIONS_API_VERSION and PUBLISHER_API_VERSION

**Review by joehan — COMMENTED** [review]
Quick drive by review

**Comment by apascal07** [comment]
@joehan The diffing is awful in the tests file but I changed `publishExtensionVersion` to `createExtensionVersionFromLocalSource`, added `createExtensionVersionFromGitHubSource`, and deleted `unpublishExtension` (along with its method, there are no references to it anymore and it's been deprecated for ages).

**Review by joehan — APPROVED** [review]
LGTM

**Pull request merged**